### ISSUE TITLE
Can't use ParseRule#getContent with a dynamic schema

### ISSUE THREAD

**Issue by 5id** [ProseMirror/prosemirror#804]
### Issue details

Currently the interface for `ParseRule#getContent` is `?⁠fn(dom.Node) → Fragment`
(https://prosemirror.net/docs/ref/#model.ParseRule.getContent)

This assumes that you have an instance of schema in the scope of the declaring file, to enable you to create a Fragment. This isn't always the case, especially if you're creating the schema outside of where you are defining the Mark/Node specs.

Extending the interface to also pass through the schema of the DOMParser would make this easier.

e.g. `ParseRule#getContent: ?⁠fn(dom.Node, Schema) → Fragment`

### ProseMirror version

prosemirror-model@1.4.4

**Comment by marijnh** [comment]
> This isn't always the case, especially if you're creating the schema outside of where you are defining the Mark/Node specs.

Since you're only going to be able to use the parser _after_ you've created the schema, I'd suggest just creating it in a function that takes the schema as an argument. Passing useful context information to callbacks everywhere is just going to be messy, and since the structure of the parser will already depend on the schema, I don't think closing over it is a very problematic requirement here.

**Comment by scottsidwell** [comment]
I think I wasnt as clear as I could have been

Say I have a file schema.js:
```javascript

...
const codeBlock = {
  ...
  parseDOM: [{
    parseContent(node) {
      return schema.text(“hello world”);
    }
  }]
};

const schema = new Schema({
  nodes: {
    text,
    doc,
    codeBlock
  }
});
export default schema;
```

That would work fine. But in my case, where  I’m creating the schema in a file *separate* to where I am defining the node/mark specs - I don’t see a way of making this work. Am I missing something here?

**Comment by marijnh** [comment]
Ah, the rule is attached to a node—yeah, that does make sense. I've made the change in attached patch and released prosemirror-model 1.5.0
